File: C1.UWP.FlexGrid/CS/EMenus/Data/CustomStyleSelector.cs
using C1.Xaml;
using Windows.UI;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;
using Windows.UI.Xaml.Media;

namespace Grapecity.C1_EMenus
{
    public class CustomStyleSelector : C1StyleSelector
    {

        protected override Style SelectStyleCore(object item, DependencyObject container)
        {
            var tvi = (C1TreeViewItem)container;
            if (tvi.Header is Category)
            {
                Style style = ((Style)Resources["ExpandedStyle"]);

                return style;
            }

            if (tvi.Header is SubCategory)
            {

                return (Style)Resources["ExpandedStyle"];
            }
            return null;
        }
    }
}

File: C1.UWP.FlexGrid/VB/EMenus/Data/CustomStyleSelector.vb
Imports C1.Xaml
Public Class CustomStyleSelector
    Inherits C1StyleSelector

    Protected Overrides Function SelectStyleCore(item As Object, container As DependencyObject) As Style
        Dim tvi As C1TreeViewItem = CType(container, C1TreeViewItem)
        If TypeOf tvi.Header Is Category Then
            Dim style As Style = (CType(Resources("ExpandedStyle"), Style))

            Return style
        End If

        If TypeOf tvi.Header Is SubCategory Then

            Return CType(Resources("ExpandedStyle"), Style)
        End If
        Return Nothing
    End Function
End Class